packages/connection-form/src/hooks/use-connect-form-preferences.tsx

```diff
@@ -29,6 +34,7 @@ const defaultPreferences = {
   showKerberosAuth: true,
   showCSFLE: true,
   showProxySettings: true,
+  saveAndConnectLabel: 'Connect',
```

Comment:
I think the intention for these preferences was only to map compass preferences to connection form without the need to have a dependency on the preferences model (although @mcasimir might correct me on that), this is just something that you'd usually pass as a component property, doesn't feel like it belongs here

Reply:
Hmm.. I put it there because 'showFavoriteActions' was already there. There is a note above that says '// Not all of these preference map to Compass preferences.', which suggests that what you say used to be true, but at this point maybe we can keep the form preferences together?

Reply:
If we're already diverging from React patterns in an unconventional ways here and don't want to touch this particular part, I guess it's okay to keep it for now like that for consistency, but just want to make sure we all understand that we are leaving more clean up to do after a "clean up" PR here

Reply:
@gribnoysup as discussed, changed the provider into ConnectFormSettingsProvider, which combines compass preferences and other settings (such as the VSCode customizations). externally, these are top level properties